app/components/autocomplete/emoji_suggestion/emoji_suggestion.js

```diff
@@ -63,18 +72,19 @@ export default class EmojiSuggestion extends Component {
             return;
         }
 
-        const matchTerm = match[3];
+        const oldMatchTerm = this.matchTerm;
+        this.matchTerm = match[3];
 
-        const matchTermChanged = matchTerm !== this.state.matchTerm;
-        if (matchTermChanged) {
-            this.setState({
-                matchTerm
-            });
+        const pre47 = !isMinimumServerVersion(Client4.getServerVersion(), 4, 7);
+
+        if (!pre47 && this.matchTerm !== oldMatchTerm && this.matchTerm.length) {
+            this.props.actions.autocompleteCustomEmojis(this.matchTerm);
+            return;
         }
 
-        if (matchTermChanged) {
-            this.handleFuzzySearch(matchTerm, nextProps);
-        } else if (!matchTerm.length) {
+        if (pre47 || this.props.emojis !== nextProps.emojis) {
```

Comment:
previously this was `if (matchTerm !== this.state.matchTerm)` why the change on the if statement?

Reply:
For the pre 4.7 case I think you're right I need to check that here